lib/macro-substitutor.js

```diff
@@ -41,19 +40,38 @@ var _MACRO_REPLACER = function(valueMap, matchedMacro, macroKey) {
  * method will find and replaces macros with values defined in the a the map.
  *
  * @param {string} template Template whose macros will be substituted.
- * @param {{
- *   id: string,
- *   bidid: string,
- *   impid: string,
- *   seat: string,
- *   adid: string,
- *   price: number,
- *   cur: string
- * }} valueMap A map of substitution values whose keys are mapped to the macro,
- *     and the respective values are to replace the macros. NOTE: Values in the
- *     map are assumed to be sanitized meaning they are already escaped/encoded.
+ * @param {!SubstitutionValueMap} valueMap A map of substitution values.
+ * @param {boolean=} opt_shouldEncode Optional boolean flag to indicate whether
+ *     the values in the substitution value map should be URI encoded.
  * @return {string} Returns fully substituted string, or throw in case of error.
  */
-module.exports = function(template, valueMap) {
-  return template.replace(MACRO_PATTERN, _MACRO_REPLACER.bind(null, valueMap));
+module.exports = function(template, valueMap, opt_shouldEncode) {
+  var newValueMap = {};
+
+  if (opt_shouldEncode) {
+    for (var key in valueMap) {
+      newValueMap[key] = encodeURIComponent(valueMap[key]);
+    }
+  } else {
+    newValueMap = valueMap;
+  }
+
+  return template.replace(
+      MACRO_PATTERN,
```

Comment:
Thanks for the feedback! For not I'm going to leave this indentation as it; please bear with me on this style obscureness for now. I want to wait until we have a style guide (draft coming soon), where we can critique on formally. Once that finalizes, I'll come back to this repo to update them.